components/datadog/dockeragentparams/params.go

```diff
@@ -136,18 +134,16 @@ func WithFakeintake(fakeintake *fakeintake.Fakeintake) func(*Params) error {
 }
 
 func withIntakeHostname(url pulumi.StringInput, shouldSkipSSLValidation bool) func(*Params) error {
-	shouldEnforceHTTPInput := url.ToStringOutput().ApplyT(func(host string) (bool, error) {
-		return strings.HasPrefix(host, "https"), nil
-	}).(pulumi.BoolOutput)
 	return func(p *Params) error {
 		envVars := pulumi.Map{
 			"DD_DD_URL":                                 pulumi.Sprintf("%s", url),
 			"DD_PROCESS_CONFIG_PROCESS_DD_URL":          pulumi.Sprintf("%s", url),
 			"DD_APM_DD_URL":                             pulumi.Sprintf("%s", url),
 			"DD_SKIP_SSL_VALIDATION":                    pulumi.Bool(shouldSkipSSLValidation),
 			"DD_REMOTE_CONFIGURATION_NO_TLS_VALIDATION": pulumi.Bool(shouldSkipSSLValidation),
+			"DD_LOGS_CONFIG_FORCE_USE_HTTP":             pulumi.Bool(true), // Force the use of HTTP/HTTPS rather than switching to TCP
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Why removing the logic of `shouldEnforceHTTPInput`, that looking at it now is inverted?

Reply:
This actually enforces the use of HTTP or HTTPS. Without this the Agent will attempt to connect to the Fake Intake and if that fails will then revert to trying to send logs via TCP. This isn't desirable behaviour (unless we are testing TCP fallback) as if it reverts to TCP it never reverts back to HTTP/HTTPS.

See [here](https://github.com/DataDog/datadog-agent/blob/42c07251e2fded462dde557410c1a80e00c5ed40/pkg/config/config_template.yaml#L941) for more info on this config option. It honestly not well named configuration option 😅

Reply:
No way, today I learned! Thank you for the explanation!